double-conversion/strtod.cc

```diff
@@ -483,7 +483,27 @@ float Strtof(Vector<const char> buffer, int exponent) {
   double double_guess;
   bool is_correct = ComputeGuess(trimmed, exponent, &double_guess);
 
-  float float_guess = static_cast<float>(double_guess);
+  // ASAN has a sanitize check that disallows casting doubles to floats if
+  // they are too big.
+  // https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#available-checks
+  // The behavior should be covered by IEEE 754, but some projects use this
+  // flag, so work around it.
+  float max_finite = 3.4028234663852885981170418348451692544e+38;
+  // The half-way point between the max-finite and infinity value.
+  // Since infinity has an even significand everything equal or greater than
+  // this value should become infinity.
+  double half_max_finite_infinity =
+      3.40282356779733661637539395458142568448e+38;
+  float float_guess;
+  if (double_guess >= max_finite) {
+    if (double_guess >= half_max_finite_infinity) {
+      float_guess = Single::Infinity();
```

Comment:
Do we need to check that f3 and f4 below don't overflow as well? (I can conceive of the case where double_guess is just on the border of overflowing, and the next double(s) up might overflow, but I'm not sure if that can actually happen.

Reply:
Would it be safer to write a `SanitizedDoubletof` function that basically has all of this logic, and is used everywhere instead of `static_cast<float>` where it might possibly overflow?

Reply:
Thanks. I could probably just shortcut the computation entirely, but I don't have the time right now to think this through.
Went for the code you wrote in your CL. It's just simpler...